tests/test_core.py

```diff
@@ -125,6 +126,23 @@ def needs_make(note=''):
   return lambda f: f
 
 
+def no_asan(note):
+  def decorator(f):
+    assert callable(f)
+
+    @wraps(f)
+    def decorated(self, *args, **kwargs):
+      if '-fsanitize=address' in self.emcc_args:
+        self.skipTest(note)
+      f(self, *args, **kwargs)
+    return decorated
+  if callable(note):
+    f = note
+    note = 'This test cannot be run under ASan'
+    return decorator(f)
+  return decorator
+
```

Comment:
why is this `no_*` method much more complicated/long than the other ones? i feel like i'm missing something, or it's needlessly long?

Reply:
Simplified the decorator.